app/models/miq_report/seeding.rb

```diff
@@ -1,76 +1,73 @@
 module MiqReport::Seeding
   extend ActiveSupport::Concern
 
-  module ClassMethods
-    REPORT_DIR  = File.expand_path(File.join(Rails.root, "product/reports"))
-    COMPARE_DIR = File.expand_path(File.join(Rails.root, "product/compare"))
+  REPORT_DIR  = Rails.root.join("product/reports")
+  COMPARE_DIR = Rails.root.join("product/compare")
 
+  module ClassMethods
     def seed
-      # Force creation of model instances for all report yaml files that exist in the product/reports directories
-      # that don't already have an instance in the model
-      MiqReport.sync_from_dir("report")
-      MiqReport.sync_from_dir("compare")
-    end
-
-    def seed_report(pattern, type = "report")
-      dir = type == "report" ? REPORT_DIR : COMPARE_DIR
-      files = Dir.glob(File.join(dir, "**/*#{pattern}*"))
-      files.collect do |f|
-        sync_from_file(f, dir, type)
+      reports = where(:rpt_type => 'Default').where.not(:filename => nil).index_by do |f|
+        seed_filename(f.filename)
       end
-    end
 
-    def sync_from_dir(typ)
-      if typ == "report"
-        dir = REPORT_DIR
-        pattern = "*/*.yaml"
-        cond = {:rpt_type => 'Default', :template_type => [typ, nil]}
-      else
-        dir = COMPARE_DIR
-        pattern = "*.yaml"
-        cond = {:rpt_type => 'Default', :template_type => typ}
+      seed_files.each do |f|
+        filename = seed_filename(f)
+        seed_record(f, filename, reports.delete(filename))
       end
 
-      where(cond).where.not(:filename => nil).each do |f|
```

Comment:
I changed this significantly, and need another set of eyes to make sure I'm doing this right.

The old code is convoluted because it does report syncing in 2 passes.  Because of this the query needs to be more complicated.  For example, when it's the report directory's turn, it has to say `"Default" but only "report" or nil typ`. 

The new code does all reports in one pass, so it doesn't need that extra check to avoid the other type, and is thus just "Default"

Reply:
I think what concerns me here is that I don't know if user reports can be "Default" or not.  @gtanzillo?